packages/design-picker/src/hooks/use-filtered-designs.ts

```diff
@@ -28,21 +28,33 @@ export const getFilteredDesignsByCategory = (
 	}
 
 	// Get designs by the selected category.
+	// Note that we don't want to show a theme in multiple sections.
+	// See https://github.com/Automattic/dotcom-forge/issues/10110.
 	const categorySlugsSet = new Set( categorySlugs );
 	for ( let i = 0; i < filteredDesigns.length; i++ ) {
 		const design = filteredDesigns[ i ];
-		let count = 0;
+		let lastMatchedCategory = null;
+		let matchedCount = 0;
 		for ( let j = 0; j < design.categories.length; j++ ) {
 			const category = design.categories[ j ];
 			if ( categorySlugsSet.has( category.slug ) ) {
-				filteredDesignsByCategory[ category.slug ].push( design );
-				count++;
+				// We show the designs for the last selected category on top first
+				// so it would be better to put the design into the last matched category
+				// if it doesn't match all selected categories.
+				lastMatchedCategory = category.slug;
+				matchedCount++;
 			}
 		}
```

Comment:
super ![badge](https://img.shields.io/badge/review-nits-green.svg): slightly improved version in performance for variable assignments 

```js
    const matchedCategories = design.categories.filter(category =>
        categorySlugsSet.has(category.slug)
    );
    const matchedCount = matchedCategories.length;

	...

    if (matchedCategories.length > 0) {
        const lastMatchedCategory = matchedCategories[matchedCategories.length - 1].slug;
        filteredDesignsByCategory[lastMatchedCategory].push(design);
    }
```

Reply:
👍